### PR TITLE
feat(client): add apply changes confirmation dialog BTM

### PR DESCRIPTION
**Pull request by morhag90**
Close: #7883 

## PR Details

 Add the apply changes confirmation dialog after re-entering the architecture tab. 
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizer.tsx [hunk 1]
```diff
@@ -284,6 +304,15 @@ export default function ModelOrganizer() {
...
           <div className={`${CLASS_NAME}__body`}>
+            <ConfirmationDialog
+              isOpen={confirmApplyChangesDialog}
+              onDismiss={onApplyChangesConfirmationClicked}
+              message={`Your architecture tweaks are ready to be applied, allowing Amplication to generate its code.
+              To reset or fetch updates not in the current state, use the buttons in the top toolbar.`}
+              confirmButton={{ label: "Got it" }}
+              onConfirm={onApplyChangesConfirmationClicked}
+            ></ConfirmationDialog>
+
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizer.tsx:315]: Use a new component <ModelOrganizerChangesExistConfirmation/> and move all the logic including the useState and messages to it to keep the parent component lean

File: packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizer.tsx [hunk 2]
```diff
@@ -98,6 +98,12 @@ export default function ModelOrganizer() {
   const [currentDropTarget, setCurrentDropTarget] = useState<Node>(null);
 
   const [isValidResourceName, setIsValidResourceName] = useState<boolean>(true);
+  const [
+    saveConfirmApplyChangesTampTrigger,
+    setSaveConfirmApplyChangesTampTrigger,
+  ] = useState<Date>(null);
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizer.tsx:104]: why do we need two useStates?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  changes: ModelChanges;
+};
+
+const ApplyChangesConfirmationModelOrganizer = ({ changes }: Props) => {
+  const [confirmApplyChangesDialog, setConfirmApplyChangesDialog] = useState<
+    boolean | null
+  >(null);
+
+  useEffect(() => {
+    if (
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx:16]: else set(false)

File: packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import { useCallback, useEffect, useState } from "react";
+import "./CreateApplyChangesLoader.scss";
+import { ModelChanges } from "./types";
+
+type Props = {
+  changes: ModelChanges;
+};
+
+const ApplyChangesConfirmationModelOrganizer = ({ changes }: Props) => {
+  const [confirmApplyChangesDialog, setConfirmApplyChangesDialog] = useState<
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx:11]: showDialog
```suggestion
  const [showDialog, setShowDialog] = useState<
```

File: packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { ConfirmationDialog } from "@amplication/ui/design-system";
+import { useCallback, useEffect, useState } from "react";
+import "./CreateApplyChangesLoader.scss";
+import { ModelChanges } from "./types";
+
+type Props = {
+  changes: ModelChanges;
+};
+
+const ApplyChangesConfirmationModelOrganizer = ({ changes }: Props) => {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ArchitectureConsole/ApplyChangesConfirmationModelOrganizer.tsx:10]:
```suggestion
const ModelOrganizerPreviousChangesExistConfirmation = ({ changes }: Props) => {
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
you need to set the state to false otherwise the message will pop after doing the first change because the state is still null

**Review by overbit — APPROVED** [review]

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**